src/shared/numa-util.c

```diff
@@ -124,6 +126,58 @@ int numa_to_cpu_set(const NUMAPolicy *policy, CPUSet *ret) {
         return 0;
 }
 
+static int numa_max_node(int *ret) {
+        _cleanup_closedir_ DIR *d = NULL;
+        struct dirent *de;
+        int r, max_node = 0;
```

Comment:
hmm, as a safety precaution I think this should be initialized to 1 actually. After all it's guaranteed we have at least one node (otherwise, what node are we ourselves running on?)

Reply:
There is no need to do that. Function returns index of largest node instead of count.